src/lucky/secure_headers/set_frame_guard.cr

```diff
@@ -0,0 +1,56 @@
+module Lucky
+  module SecureHeaders
+    # This module sets the HTTP header [X-Frame-Options](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options).
+    # It's job is responsible for deciding which site can call your site from within a frame.
+    # For more information, read up on [Clickjacking](https://en.wikipedia.org/wiki/Clickjacking).
+    #
+    # Include this module in the actions you want to add this to.
+    # A required method `frame_guard_options` must be defined`
+    # ```
+    # class BrowserAction < Lucky::Action
+    #   include Lucky::SecureHeaders::SetFrameGuard
+    #
+    #   def frame_guard_options
+    #     {allow_from: "deny"}
+    #   end
+    # end
+    # ```
+    #
+    # ### Options
+    # The `frame_guard_options` method must be defined and return `NamedTuple(allow_from: String)`
+    # The `allow_from` key can have one of 3 String values:
+    # - `"same"` or `"sameorigin"`
+    # - `"nowhere" or `"deny"`
+    # - a valid URL e.g. `"https://mysite.com"`
+    module SetFrameGuard
+      macro included
+        before set_frame_guard_header
+      end
+
+      abstract def frame_guard_value : String
+
+      private def set_frame_guard_header
+        context.response.headers["X-Frame-Options"] = check_frame_guard_value!(frame_guard_value)
+        continue
+      end
+
+      private def check_frame_guard_value!(value : String)
+        v = value.downcase
+        case v
+        when "sameorigin", "deny"
+          v
+        when /^https?:\/\/[0-9a-zA-Z_-]+\.?[0-9a-zA-Z_-].+$/
```

Comment:
I am nervous that this Regex might be too strict. I don't really know, but it could bite us in the future by disallowing a valid URL. What do you think about instead checking if the first characters start with `http` and leave whether it is valid up. to the user. This will still catch the most common errors (misspelled sameorigin, or some total incorrect value), but still let people use whatever URL they want. 

Thoughts?

Reply:
Makes sense. If it's close enough it should be fine.